src/pybamm/config.py

```diff
@@ -45,20 +48,125 @@ def is_running_tests():  # pragma: no cover
+        try:
+            from IPython.display import clear_output
+
+            user_input = input("Do you want to enable telemetry? (Y/n): ")
+            clear_output()
+
+            return user_input
+
+        except Exception:  # pragma: no cover
+            return None
+
+    # 2. Windows-specific handling
+    if sys.platform == "win32":
+        try:
+            import msvcrt
+
+            start_time = time.time()
+            input_chars = []
+            sys.stdout.write("Do you want to enable telemetry? (Y/n): ")
+            sys.stdout.flush()
+
+            while time.time() - start_time < timeout:
+                if msvcrt.kbhit():
+                    char = msvcrt.getwche()
+                    if char in ("\r", "\n"):
+                        sys.stdout.write("\n")
+                        return "".join(input_chars)
+                    input_chars.append(char)
+                time.sleep(0.1)
+            return None
+        except Exception:
+            return None
 
-    Returns
-    -------
-    bool
-        True if the user opts in, False otherwise.
+    # 3. POSIX-like systems will need to use termios
+    else:  # pragma: no cover
+        try:
+            import termios
+            import tty
+            import select
+
+            # Save terminal settings for later
+            old_settings = termios.tcgetattr(sys.stdin)
+            try:
+                # Set terminal to raw mode
+                tty.setraw(sys.stdin.fileno())
+
+                sys.stdout.write("Do you want to enable telemetry? (Y/n): ")
+                sys.stdout.flush()
+
+                input_chars = []
+                start_time = time.time()
+
+                while time.time() - start_time < timeout:
+                    rlist, _, _ = select.select([sys.stdin], [], [], 0.1)
+                    if rlist:
+                        char = sys.stdin.read(1)
+                        if char in ("\r", "\n"):
+                            sys.stdout.write("\n")
+                            return "".join(input_chars)
+                        input_chars.append(char)
+                        sys.stdout.write(char)
+                        sys.stdout.flush()
+                return None
+
+            finally:
+                # Restore saved terminal settings
+                termios.tcsetattr(sys.stdin, termios.TCSADRAIN, old_settings)
+                sys.stdout.write("\n")
+                sys.stdout.flush()
+
+        except Exception:  # pragma: no cover
+            return None
+
+    return None
+
+
+def ask_user_opt_in(timeout=10):  # pragma: no cover
+    """
+    Ask the user if they want to opt in to telemetry.
     """
+    # Check for telemetry disable flag first
+    if os.getenv("PYBAMM_DISABLE_TELEMETRY", "false").lower() != "false":
+        return False
+
+    # Skip telemetry prompt in non-interactive environments
+    if not (sys.stdin.isatty() or is_notebook()):
+        return False
```

Comment:
These checks are also in the `get_input_or_timeout()` function, do we need to check that twice?

Reply:
Removed